tests/system/common.py

```diff
@@ -351,14 +351,15 @@ def install_enterprise_cli_package():
     stdout, stderr, return_code = shakedown.run_dcos_command(cmd, raise_on_error=True)
 
 
+@retrying.retry(wait_fixed=1000, stop_max_attempt_number=10, retry_on_exception=ignore_exception)
```

Comment:
What was the error that `cli` reported? Why do we need to retry here?

Reply:
The node is some times not accessible. I cannot think of anything else that could be wrong (apart from the fact that the cluster is in a weird state):
```
》dcos package list --json

 Connection refused: master.mesos/10.10.0.166:8443 at remote address: master.mesos/10.10.0.166:8443. Remote Info: Not Available
 1
package list command returned code:1, stderr:Connection refused: master.mesos/10.10.0.166:8443 at remote address: master.mesos/10.10.0.166:8443. Remote Info: Not Available
, stdout: 
```

Reply:
Search it here: https://jenkins.mesosphere.com/service/jenkins/view/Marathon-Dashboard/job/system-integration-tests/job/marathon-si-dcos-strict/job/master/2155/console

Reply:
`Connection refused: master.mesos/10.10.0.166:8443` indicates that `cli` is trying to connect to Marathon and fails. Looks like this test is somehow affected by the previous test which abdicated marathon leader. And though we're waiting for Marathon to become available on all masters before each test that seems to not be enough? 🤔 
Retrying seems like an easy fix but it is likely to mask the actual issue underneath. Nevertheless, we might merge it.

Reply:
If an agent is down, this _cannot_ be marathon's fault. Therefore in the context of "marathon" system tests, such an error is irrelevant.

In this logic, if the agent is *temporarily* inaccessible could be a recoverable DC/OS problem, which is totally expected in a distributed system. This "temporarily" check is implemented using the "retry" statement...

I **DO** agree that such an error is important on the overall DC/OS product and therefore must be forwarded to the respective team. And thus it **is** important to be able to collect enough logs for further investigation. But again, this is not a marathon-related issue.